features/gm.test.ts

```diff
@@ -237,3 +239,76 @@ Deno.test("need to add group before handling relevant messages", async () => {
         }
     }
 });
+
+Deno.test("prepare gruop message event should return correct", async () => {
+    const user_A = InMemoryAccountContext.Generate();
+    const gm_A = new GroupMessageController(user_A, { add: (_) => {} }, { add: (_) => {} });
+    const groupChat = gm_A.createGroupChat();
+
+    {
+        // blob
+        const blob = new Blob();
+        const blobMessageEvent = await gm_A.prepareGroupMessageEvent(groupChat.groupKey.publicKey, blob);
+        if (blobMessageEvent instanceof Error) fail(blobMessageEvent.message);
+
+        const decryptedContent = await groupChat.cipherKey.decrypt(
+            blobMessageEvent.pubkey,
+            blobMessageEvent.content,
+        );
+        if (decryptedContent instanceof Error) {
+            fail(decryptedContent.message);
+        }
+        const json = parseJSON<unknown>(decryptedContent);
+        if (json instanceof Error) {
+            fail(json.message);
+        }
+
+        let message;
+        try {
+            message = z.object({
+                type: z.string(),
+                text: z.string(),
+                kind: z.string(),
+            }).parse(json);
+        } catch (e) {
+            fail(e);
+        }
```

Comment:
you should not implement the decryption/decoding in the test. Instead, you should call the "get conversation message" method of the controller and only test public interfaces.

Reply:
right